sensor/kubernetes/eventpipeline/resolver/resolver_test.go

```diff
@@ -0,0 +1,538 @@
+		return &storage.Deployment{
+			Namespace: "example",
+			Labels:    map[string]string{"app": "a"},
+		}
+	})
+
+	s.mockRBACStore.EXPECT().GetPermissionLevelForDeployment(gomock.Any()).AnyTimes().
+		DoAndReturn(func(arg0 interface{}) storage.PermissionLevel { return storage.PermissionLevel_NONE })
+
+	s.mockServiceStore.EXPECT().GetExposureInfos(gomock.Any(), gomock.Any()).Times(1).
+		DoAndReturn(func(arg0, arg1 interface{}) []map[service.PortRef][]*storage.PortConfig_ExposureInfo { return exposure })
+
+	s.mockDeploymentStore.EXPECT().BuildDeploymentWithDependencies(
+		gomock.Eq(deployment), gomock.Eq(store.Dependencies{
+			PermissionLevel: storage.PermissionLevel_NONE,
+			Exposures:       exposure,
+		})).
+		Times(1).
+		DoAndReturn(func(arg0, arg1 interface{}) (*storage.Deployment, error) {
+			return &storage.Deployment{
+				Id:                            deployment,
+				ServiceAccountPermissionLevel: storage.PermissionLevel_NONE,
+			}, nil
+		})
+}
+
+func (s *resolverSuite) givenAnyDeploymentProcessedNTimes(times int) {
+	s.mockDeploymentStore.EXPECT().Get(gomock.Any()).Times(times).DoAndReturn(func(arg0 interface{}) *storage.Deployment {
+		return &storage.Deployment{}
+	})
+
+	s.mockRBACStore.EXPECT().GetPermissionLevelForDeployment(gomock.Any()).Times(times).
+		DoAndReturn(func(arg0 interface{}) storage.PermissionLevel { return storage.PermissionLevel_DEFAULT })
+
+	s.mockServiceStore.EXPECT().GetExposureInfos(gomock.Any(), gomock.Any()).Times(times).
+		DoAndReturn(func(arg0, arg1 interface{}) []map[service.PortRef][]*storage.PortConfig_ExposureInfo { return nil })
+
+	s.mockDeploymentStore.EXPECT().BuildDeploymentWithDependencies(gomock.Any(), gomock.Any()).
+		Times(times).
+		DoAndReturn(func(arg0, arg1 interface{}) (*storage.Deployment, error) {
+			return &storage.Deployment{}, nil
+		})
+}
+
+type deploymentMatcher struct {
+	id              string
+	permissionLevel storage.PermissionLevel
+	exposure        interface{}
+	error           string
+}
+
+func (m *deploymentMatcher) Matches(target interface{}) bool {
+	event, ok := target.(*component.ResourceEvent)
+	if !ok {
+		m.error = "received message isn't a resource event"
+		return false
+	}
+
+	if len(event.ForwardMessages) < 1 {
+		m.error = fmt.Sprintf("not enough ForwardMessages: %d", len(event.ForwardMessages))
+		return false
+	}
+
+	deployment := event.ForwardMessages[0].GetDeployment()
+	if deployment == nil {
+		m.error = "no deployment in resource event message"
+		return false
+	}
+
+	if deployment.GetId() != m.id {
+		m.error = fmt.Sprintf("IDs don't match: expected %s != %s", m.id, deployment.GetId())
+		return false
+	}
+
+	if deployment.GetServiceAccountPermissionLevel() != m.permissionLevel {
+		m.error = fmt.Sprintf("Permission level doesn't match %s != %s", m.permissionLevel, deployment.GetServiceAccountPermissionLevel())
+		return false
+	}
+
```

Comment:
I think the `deploymentMatcher` should also check the `Ports`

Reply:
Done